gcd/src/test/java/org/spine3/server/storage/datastore/BigDataTester.java

```diff
@@ -0,0 +1,165 @@
+        return this;
+    }
+
+    public BigDataTester setWriteLimit(long writeMillisLimit) {
+        this.writeMillisLimit = writeMillisLimit;
+        return this;
+    }
+
+    public BigDataTester setReadLimit(long readMillisLimit) {
+        this.readMillisLimit = readMillisLimit;
+        return this;
+    }
+
+    /**
+     * Preforms the actual check.
+     *
+     * <p>The execution flow is as follows:
+     * <ol>
+     *     <li>1. Produce the records with the given {@link EntrySupplier}
+     *     <li>2. Measure the time of the {@linkplain RecordStorage#write(Map) bulk write}
+     *     <li>3. Fail if the time is over the specified limit
+     *     <li>4. Wait 1 second to ensure the Datastore has established the data consistency
+     *     <li>5. Measure the time of the {@linkplain RecordStorage#readAll() bulk read}
+     *     <li>6. Fail if the time is over the specified limit
+     *     <li>7. Check the count of the records written and read is equal
+     * </ol>
+     *
+     * <p>This method performs {@code debug} logging of the measure results. To see the log, run
+     * the tests with {@code debug} logging.
+     */
+    public void testBigDataOperations() {
+        final Map<I, EntityRecordWithColumns> records = new HashMap<>(bulkSize);
+        for (int i = 0; i < bulkSize; i++) {
+            records.put(entrySupplier.newId(), entrySupplier.newRecord());
+        }
+
+        final long writeStart = System.currentTimeMillis();
+        storage.write(records);
+        final long writeEnd = System.currentTimeMillis();
+
+        final long writeTime = writeEnd - writeStart;
+        if (writeTime > writeMillisLimit) {
+            fail(format("Writing took too long. Expected %d millis but was %d millis.", writeMillisLimit, writeTime));
+        }
+
+        log().debug("Writing took {} millis.", writeTime);
+
+        try {
+            Thread.sleep(1000);
+        } catch (InterruptedException e) {
+            fail(Throwables.getStackTraceAsString(e));
+        }
+
+        final long readStart = System.currentTimeMillis();
+
+        // Do not test data equality here, only the sizes and time
+        final Map<I, EntityRecord> readRecords = storage.readAll();
+
+        final long readEnd = System.currentTimeMillis();
+        final long readTime = readEnd - readStart;
+
+        if (readTime > readMillisLimit) {
+            fail(format("Reading took too long. Expected %d millis but was %d millis.", readMillisLimit, readTime));
+        }
+        log().debug("Reading took {} millis.", readTime);
+
+        assertEquals("Unexpected records count read.", records.size(), readRecords.size());
+    }
+
+    /**
+     * A supplier of the records to write into the Datastore for the check.
+     *
+     * <p>It's recommended to provide generic non-empty records to make the check closer to a real-life cases.
+     *
+     * @param <I> the type of the record ID
+     */
+    public interface EntrySupplier<I> {
```

Comment:
Please check [Guava `Supplier` interface](https://google.github.io/guava/releases/20.0/api/docs/com/google/common/base/Supplier.html) and rethink the suppliers you have.